src/main/java/net/neoforged/neoforge/capabilities/RegisterCapabilitiesEvent.java

```diff
@@ -79,6 +79,18 @@ public boolean isBlockRegistered(BlockCapability<?, ?> capability, Block block)
         return capability.providers.containsKey(block);
     }
 
+    /**
+     * Makes a block capability non-proxyable.
+     *
+     * <p>This method should only be called by the mod that defines the capability.
+     * Do not call this method if this is not explicitly part of the contract of the capability.
+     *
+     * @see BlockCapability#isProxyable()
+     */
+    public void setNonProxyable(BlockCapability<?, ?> capability) {
```

Comment:
Is this possible to be made to be chained so that we could call something like
```java
    BlockCapability<SomeClass, Void> BLOCK = BlockCapability.createVoid(resource("some_handler"), SomeClass.class).nonProxyable();

```

Reply:
I'm not a fan for the reason that it is dependent on classloading, and might be called at any time.